python/ray/node.py

```diff
@@ -224,10 +224,10 @@ def __init__(self,
                 ray_constants.GCS_PORT_ENVIRONMENT_VARIABLE)
             if gcs_server_port:
                 ray_params.update_if_absent(gcs_server_port=gcs_server_port)
-            if ray_params.gcs_server_port is None:
+            if ray_params.gcs_server_port is None \
+                    or ray_params.gcs_server_port <= 0:
```

Comment:
Shouldn't we just throw an exception here?

Reply:
Ray does not validate any port number right now. We can add a validation for all port flags later. Updated to just comparison.